### PR TITLE
Getting started example for write behind #253

### PR DESCRIPTION
**Pull request by AbfrmBlr**
This needs to go in for #253 to complete

### CODE REVIEW

**Comment by lorban** [comment]
This sample is good but not very visual as the WriteBehindTestLoaderWriter loader/writer does all its magic internally without logging anything.

Maybe using an inline CacheLoaderWriter that extends from WriteBehindTestLoaderWriter to add some INFO logs would be more useful for a GettingStarted example? Or do you guys think that would be pointless / over the top?

File: impl/src/test/java/org/ehcache/GettingStarted.java [hunk 1]
```diff
@@ -243,4 +302,82 @@ public void onEvent(CacheEvent<Object, Object> event) {
...
+    @Override
+    public Map<K, V> loadAll(Iterable<? extends K> keys) throws Exception {
+      throw new UnsupportedOperationException("Implement me!");
+    }
+
+    @Override
+    public void write(K key, V value) throws Exception {
+      lock.writeLock().lock();
+      try {
+        LOGGER.info("Key - '{}' ,Value - '{}' successfully wriiten", key, value);
```
Comment (ljacomet) [inline impl/src/test/java/org/ehcache/GettingStarted.java]: Typo: double `i`

File: core/src/main/java/org/ehcache/config/writebehind/WriteBehindConfigurationBuilder.java [hunk 2]
```diff
@@ -58,19 +58,31 @@ public WriteBehindConfigurationBuilder segment(int concurrency) {
...
   }
   
-  public WriteBehindConfigurationBuilder batch(int batchSize) {
+  public WriteBehindConfigurationBuilder disableCoalescing() {
+    this.writeCoalescing = false;
+    
+    return this;
+  }
+  
+  public WriteBehindConfigurationBuilder enableBatching(int batchSize) {
```
Comment (ljacomet) [inline core/src/main/java/org/ehcache/config/writebehind/WriteBehindConfigurationBuilder.java]: I would revert the change with regards to batching. `batch(1)` was a good way of saying no watching. It would require the logic to be improved though - including refusing values 0 or below.

**Comment by AbfrmBlr** [comment]
test this please

File: core/src/main/java/org/ehcache/config/writebehind/WriteBehindConfiguration.java [hunk 3]
```diff
@@ -36,21 +36,52 @@
...
+    if(writeBehindMaxQueueSize < 0) {
+      throw new IllegalArgumentException("WriteBehind queue size cannot be less than 0.");
+    }
+    if(retryAttempts < 0) {
+      throw new IllegalArgumentException("RetryAttempts cannot be less than 0.");
+    }
+    if(retryAttemptDelaySeconds < 1) {
+      throw new IllegalArgumentException("RetryAttemptDelaySeconds cannot be less than 1.");
+    }
+    if(writeBatchSize < 0) {
```
Comment (ljacomet) [inline core/src/main/java/org/ehcache/config/writebehind/WriteBehindConfiguration.java]: Can't even be less than one
